ui/selectors/selectors.js

```diff
@@ -3069,3 +3072,65 @@ export const getTokenBalancesEvm = createDeepEqualSelector(
     return tokensWithBalance;
   },
 );
+
+export const getMultiChainAssets = createDeepEqualSelector(
+  (_state, selectedAccount) => selectedAccount,
+  getMultichainBalances,
+  getAccountAssets,
+  getAssetsMetadata,
+  getAssetsRates,
+  (_state) => _state.metamask.conversionRates,
+  (
+    selectedAccountAddress,
+    multichainBalances,
+    accountAssets,
+    assetsMetadata,
+    assetRates,
+    multichainCoinRates,
+  ) => {
+    const assetIds = accountAssets?.[selectedAccountAddress.id] || [];
+    const balances = multichainBalances?.[selectedAccountAddress.id];
+    return assetIds.map((assetId) => {
+      const [chainId, assetDetails] = assetId.split('/');
+      const isToken = assetDetails.split(':')[0] === 'token';
```

Comment:
I would not do it this way, the `token` part of the Solana CAIP-19 namespace is only for Solana, if you want to differentiate between a native asset and a token you should look for SLIP-44, native assets always use the SLIP-44 namespace

Reply:
For reference: https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-20.md

Reply:
Okay thanks for this feedback. I'll update this now

Reply:
https://github.com/MetaMask/metamask-extension/pull/30389/commits/c8dbe2c91a3415b6fd78e8b67fc57cc48265095d